stlab/concurrency/main_executor.hpp

```diff
@@ -104,7 +104,28 @@ struct main_executor_type {
 
 #elif STLAB_MAIN_EXECUTOR(EMSCRIPTEN)
 
-using main_executor_type = default_executor_type;
+struct main_scheduler_type {
+    using result_type = void;
+
+    template <class F>
+    void operator()(F&& f) const {
+        using function_type = typename std::remove_reference<F>::type;
+        auto p = new function_type(std::forward<F>(f));
+
+        emscripten_async_run_in_main_runtime_thread(
+            EM_FUNC_SIG_VI,
+            static_cast<void(*)(void*)>([](void* f_) {
+                emscripten_async_call(
+                    [](void* f_) {
+                        auto f = static_cast<function_type*>(f_);
+                        (*f)();
```

Comment:
If `*f` throws an exception then there's a memory leak, no?

Reply:
I need to point you to @sean-parent for that. I did not write this code.

Reply:
That said - I agree some amount of exception handling is likely required here.

Reply:
Operations queued to the task system cannot throw as a precondition. We use packaged tasks to marshal exceptions.

Reply:
@sean-parent I'll add a comment in the source to that effect